src/components/admin/form-components/withFalcorSearch.jsx

```diff
@@ -42,49 +39,40 @@ export const withSearchableConcept = (
           [
             [
               'search',
-              category,
+              mode,
               query,
               { length: this.props.length },
-              _.uniq([...fields, ...this.props.fields, 'id']),
+              _.uniq([...fields, ...this.props.fields, 'id', 'slug']),
             ],
           ],
         );
 
-        this.props.falcor.get(...pathSets).then(x => {
-          if (!x) {
-            this.setState({ suggestions: [] });
-            return;
-          }
-          x = cleanupFalcorKeys(x); // eslint-disable-line no-param-reassign
-          const suggestions = _.toArray(x.json.search[category][query])
-            .filter(item => item !== undefined)
-            .map(item => ({
-              title: formatter(item),
-              id: item.id,
-              slug: item.slug,
-            }));
-          this.setState({ suggestions });
+        return new Promise(resolve => {
```

Comment:
You actually don't need to create a new promise here, `.then` returns a promise already so you can just return the statement